storage/file.go

```diff
@@ -114,6 +132,24 @@ func (f *File) Create(maxSize uint64) error {
 	return nil
 }
 
+// CopyFile operation copied a file/blob from the sourceURL to the path provided.
+//
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/copy-file
+func (f File) CopyFile(sourceURL string, options *FileRequestOptions, metadata map[string]string) error {
+	extraHeaders := map[string]string{
+		"x-ms-type":        "file",
+		"x-ms-copy-source": sourceURL,
+	}
+
+	headers, err := f.fsc.createResource(f.buildPath(), resourceFile, options.getParameters(), mergeMDIntoExtraHeaders(f.Metadata, extraHeaders))
```

Comment:
Have a question around this one.  I modified createResource and createResourceNoClose to accept the url.Values as parameters since all the various create functions can technically pass in a timeout. They don't do it YET but the spec says they can so was thinking they might be added later. 

I was picturing that copying a file is creating a new file hence reusing the createResource function. Given the HTTP return code is different (201 vs 202) is there a preference to how this is handled?

I could create another method similar to createFunction that calls createFunctionNoClose but simply checks for 202 instead of 201 OR I could make the expected status code to just be a parameter.

Either works... one is duplicated code (yuck) and the other is parameter with expected status (still maybe yuck).

I know it's a small/simple problem, but just wondering if there is a preferred approach here?